src/Microsoft.ML.StandardLearners/Standard/MultiClass/Ova.cs

```diff
@@ -67,72 +103,98 @@ protected override OvaPredictor TrainCore(IChannel ch, RoleMappedData data, int
             for (int i = 0; i < predictors.Length; i++)
             {
                 ch.Info($"Training learner {i}");
-                predictors[i] = TrainOne(ch, GetTrainer(), data, i);
+                predictors[i] = TrainOne(ch, GetTrainer(), data, i).Model;
             }
-            return OvaPredictor.Create(Host, Args.UseProbabilities, predictors);
+            return OvaPredictor.Create(Host, _args.UseProbabilities, predictors);
         }
 
-        private TScalarPredictor TrainOne(IChannel ch, TScalarTrainer trainer, RoleMappedData data, int cls)
+        private IPredictionTransformer<TScalarPredictor> TrainOne(IChannel ch, TScalarTrainer trainer, RoleMappedData data, int cls)
         {
-            string dstName;
-            var view = MapLabels(data, cls, out dstName);
+            var view = MapLabels(data, cls);
 
-            var roles = data.Schema.GetColumnRoleNames()
-                .Where(kvp => kvp.Key.Value != CR.Label.Value)
-                .Prepend(CR.Label.Bind(dstName));
-            var td = new RoleMappedData(view, roles);
+            string trainerLabel = data.Schema.Label.Name;
 
             // REVIEW: In principle we could support validation sets and the like via the train context, but
             // this is currently unsupported.
-            var predictor = trainer.Train(td);
+            var transformer = trainer.Fit(view);
 
-            if (Args.UseProbabilities)
+            if (_args.UseProbabilities)
             {
-                ICalibratorTrainer calibrator;
-                if (Args.Calibrator == null)
-                    calibrator = null;
-                else
-                    calibrator = Args.Calibrator.CreateComponent(Host);
-                var res = CalibratorUtils.TrainCalibratorIfNeeded(Host, ch, calibrator, Args.MaxCalibrationExamples,
-                    trainer, predictor, td);
-                predictor = res as TScalarPredictor;
-                Host.Check(predictor != null, "Calibrated predictor does not implement the expected interface");
+                var calibratedModel = transformer.Model as TScalarPredictor;
+
+                // the validations in the calibrator check for the feature column, in the RoleMappedData
+                var trainedData = new RoleMappedData(view, label: trainerLabel, feature: transformer.FeatureColumn);
+
```

Comment:
@Zruty0, @tfinley@gmail.com is there a way to restore all the roles from the trainers?

The calibrator training below looks for them.